### ISSUE TITLE
trainable false

### ISSUE THREAD

**Issue by pribadihcr** [ppwwyyxx/tensorpack#482]
Hi @ppwwyyxx,

How to set conv2D variable weight become trainable=false

since I want to use clip gradient and do apply_gradient. I got error None can not iterable, when use tf.stop_gradient() to freeze varaible.

Thanks.

**Comment by ppwwyyxx** [comment]
I guess the real question is how do you clip the gradients.
If you could just remove the None values before clipping you won't have the problem.

**Comment by pribadihcr** [comment]
thanks it works
